packages/atlas/src/views/viewer/NftsView/NftsView.tsx

```diff
@@ -78,20 +78,25 @@ export const NftsView: FC = () => {
 
   const nftsWithPlaceholders = [...(displayedNfts || []), ...placeholderItems]
 
-  useEffect(() => {
-    if (!fetchMore || !nfts?.length || !totalCount) {
-      return
-    }
-    if (totalCount <= nfts.length) {
-      return
-    }
-
-    if (currentPage * tilesPerPage + tilesPerPage > nfts.length) {
-      fetchMore({
-        variables: { ...variables, after: pageInfo?.endCursor },
-      })
-    }
-  }, [currentPage, nfts, fetchMore, pageInfo, totalCount, variables, tilesPerPage])
+  const checkForRefetch = useCallback(
+    async (pageToCheck: number) => {
+      if (!fetchMore || !nfts?.length || !totalCount) {
+        return
+      }
+      if (totalCount <= nfts.length) {
+        return
+      }
+
+      const dataNeededForPage = pageToCheck * tilesPerPage + tilesPerPage
+
+      if (dataNeededForPage > nfts.length) {
+        fetchMore({
+          variables: { ...variables, after: pageInfo?.endCursor, first: dataNeededForPage - nfts.length },
```

Comment:
We don't want to fetch all the nfts when you go to the last page.
Here's the request, when you click the last page. It took 16 seconds to load:
![image](https://user-images.githubusercontent.com/51168865/223952944-feabda93-886d-44bc-a6ac-53c52fc17d42.png)
![image](https://user-images.githubusercontent.com/51168865/223953694-8825ed5b-7495-41bd-9c4a-7fe8ac41a58b.png)

I think we should use here `ownedNfts` because it allows offset-base pagination.

I'm wondering why we used the cursor-based pagination here initially, though. I think there was a reason why ownedNftsConnection is used here, but can't recall why 🤔

Reply:
Oh, my bad. I didn't notice that you were aware of the pagination issue and you already created the issue: https://github.com/Joystream/atlas/issues/3908 

Anyway, should we apply the fix that you include in this PR anyway? I mean, It kinda fixes the issue, but it's just a workaround(and this workaround introduces another issue). Should we just move to offset-based pagination already? 
@attemka I need another pair of eyes here, wdyt?